### DISCUSSION TITLE
Duda Pte 1: tipo de estructura de datos

### DISCUSSION THREAD

**Discussion by Jairuru** [IIC2133-PUC/2024-1#99] [Tarea 1] [answered]
Hola, entiendo gracias a lo que dice el enunciado que la intención es que se modele el problema utilizando heaps, pero me hace mucho más sentido hacerlo con listas ligadas. Cuál es la problemática de hacerlo de esta forma? Tiene que ver con la complejidad de las inserciones y extracciones desde los heaps?
Gracias de antemano!

**Comment by sebacarrascop** [comment, marked as answer]
Es como dices!, si bien en ciertos contextos puede ser más útil usar listas ligadas, en este caso estás trabajando directamente con prioridades (max y min), por lo que tareas como busqueda y eliminación en una lista ligada es mucho más costoso O(n) en comparación a la eliminación del elemento min o max respectivamente O(logn).
